tests/puppeteer/pages/BO/catalog/products/index.js

```diff
@@ -45,15 +65,126 @@ module.exports = class Product extends BOBasePage {
+    return this.getNumberFromText(this.productsListTableColumnID.replace('%ROW', row));
+  }
+
+  /**
+   * Get Product Name
+   * @param row
+   * @return {Promise<textContent>}
+   */
+  async getProductNameFromList(row) {
+    return this.getTextContent(this.productsListTableColumnName.replace('%ROW', row));
+  }
+
+  /**
+   * Get Product Reference
+   * @param row
+   * @return {Promise<textContent>}
+   */
+  async getProductReferenceFromList(row) {
+    return this.getTextContent(this.productsListTableColumnReference.replace('%ROW', row));
+  }
+
+  /**
+   * Get Product Category
+   * @param row
+   * @return {Promise<textContent>}
+   */
+  async getProductCategoryFromList(row) {
+    return this.getTextContent(this.productsListTableColumnCategory.replace('%ROW', row));
+  }
+
+  /**
+   * Filter price Min - Max
+   * @param min
+   * @param max
+   * @return {Promise<void>}
+   */
+  async filterPriceProducts(min, max) {
+    await this.page.type(this.productFilterPriceMinInput, min.toString());
+    await this.page.type(this.productFilterPriceMaxInput, max.toString());
+    await this.clickAndWaitForNavigation(this.filterSearchButton);
+  }
 
   /**
-   * Filter products from inputs : Name, reference and Category
+   * Get Product Price
+   * @param row
+   * @return {Promise<textContent>}
+   */
+  async getProductPriceFromList(row) {
+    return this.getNumberFromText(this.productsListTableColumnPrice.replace('%ROW', row));
+  }
+
+  /**
+   * Filter Quantity Min - Max
+   * @param min
+   * @param max
+   * @return {Promise<void>}
+   */
+  async filterQuantityProducts(min, max) {
+    await this.page.type(this.productFilterQuantityMinInput, min.toString());
+    await this.page.type(this.productFilterQuantityMaxInput, max.toString());
+    await this.clickAndWaitForNavigation(this.filterSearchButton);
+  }
+
+  /**
+   * Get Product Quantity
+   * @param row
+   * @return {Promise<textContent>}
+   */
+  async getProductQuantityFromList(row) {
+    return this.getNumberFromText(this.productsListTableColumnQuantity.replace('%ROW', row));
+  }
+
+  /**
+   * Get Product Status
+   * @param row
+   * @return {Promise<textContent>}
+   */
+  async getProductStatusFromList(row) {
+    return this.getTextContent(this.productsListTableColumnStatus.replace('%ROW', row));
+  }
+
+  /**
+   * Filter products
    * @param filterBy
    * @param value
+   * @param filterType
    * @return {Promise<void>}
    */
-  async filterProducts(filterBy, value = '') {
-    await this.page.type(this.productFilterInput.replace('%FILTERBY', filterBy), value);
+  async filterProducts(filterBy, value = '', filterType = 'input') {
+    switch (filterType) {
+      case 'input':
+        await this.page.type(this.productFilterInput.replace('%FILTERBY', filterBy), value);
+        break;
+      case 'select':
+        await this.selectByVisibleText(this.productFilterSelect.replace('%FILTERBY', filterBy),
+          value ? 'Active' : 'Inactive',
```

Comment:
So, `status` in file demo/products should be true or false instead of 'Active' or 'Inactive'